OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -30,6 +33,7 @@ public Attack(Actor self, Target target, WDist minRange, WDist maxRange, bool al
 			Target = target;
 			this.minRange = minRange;
 			this.maxRange = maxRange;
+			rangeModifiers = self.TraitsImplementing<IRangeModifier>();
```

Comment:
You should append the ```.Select(m => m.GetRangeModifier())``` to this line too, and then you can drop it from ```MaxRange```. Saves you repeating the work on every call to ```MaxRange```.